lib/file_mutex.rb

```diff
@@ -56,6 +57,10 @@ def lock!(timeout_duration = DEFAULT_TIMEOUT)
     end
     return result
   rescue Errno::EEXIST
+    File.open(@path) do |f|
+      # If we can lock the file the original process must have died - unlock now
```

Comment:
(not sure if my previous comment got lost, here is it again)
The existing code has another race condition: a successful lock will call unlock twice, one in timeout block and one in lock! block. The second unlock! may accidentally delete the file locked by other process.